auth/tokenizer.go

```diff
@@ -0,0 +1,135 @@
+package auth
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"os"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/pborman/uuid"
+	"gitlab.com/LUSHDigital/soa/first-class/api-gateway/service/config"
+)
+
+// Tokeniser is the auth tokeniser for JSON Web Tokens
+type Tokeniser struct {
+	privateKey *rsa.PrivateKey
+	publicKey  *rsa.PublicKey
+	authIssuer string
+}
+
+// NewMockTokeniser creates a new tokeniser with a random key pair
+func NewMockTokeniser() (*Tokeniser, error) {
+	reader := rand.Reader
+	bitSize := 2048
+	privateKey, err := rsa.GenerateKey(reader, bitSize)
+	if err != nil {
+		return nil, err
+	}
+	publicKey := &privateKey.PublicKey
+	name, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+	return &Tokeniser{
+		privateKey: privateKey,
+		publicKey:  publicKey,
+		authIssuer: name,
+	}, nil
+}
+
+// NewPublicTokeniser returns a new JWT instance
+func NewPublicTokeniser(bPublicKey string) (*Tokeniser, error) {
+	publicKey, err := jwt.ParseRSAPublicKeyFromPEM([]byte(bPublicKey))
+	if err != nil {
+		return nil, err
+	}
+	return &Tokeniser{publicKey: publicKey}, nil
+}
+
+// NewTokeniser returns a new JWT instance
+func NewTokeniser(bPrivateKey, bPublicKey, issuer string) (*Tokeniser, error) {
+	privateKey, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(bPrivateKey))
+	if err != nil {
+		return nil, err
+	}
+	publicKey, err := jwt.ParseRSAPublicKeyFromPEM([]byte(bPublicKey))
+	if err != nil {
+		return nil, err
+	}
+	return &Tokeniser{
+		privateKey: privateKey,
+		publicKey:  publicKey,
+		authIssuer: issuer,
+	}, nil
+}
+
+// GenerateToken generates and returns an authentication token.
+func (t *Tokeniser) GenerateToken(consumer *Consumer) (token string, err error) {
+	// Create our claims
+	// NOTE: The consumer is sanitised
+	claims := Claims{
+		Consumer: *consumer,
+		StandardClaims: jwt.StandardClaims{
+			ExpiresAt: time.Now().Add(time.Duration(config.TokenValidPeriod) * time.Minute).Unix(),
+			Issuer:    t.authIssuer,
+			Id:        uuid.New(),
+		},
+	}
+	// Create the token
+	newToken := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+	// Sign the token
+	token, err = newToken.SignedString(t.privateKey)
```

Comment:
Style nit, but drop the named returns and just `return newToken.SignedString(t.privateKey)`